astropy/cosmology/tests/test_cosmology.py

```diff
@@ -1233,11 +1234,16 @@ def test_angular_diameter_distance_z1z2():
     assert allclose(tcos.angular_diameter_distance_z1z2(1, 2),
                     646.22968662822018 * u.Mpc)
 
-    z1 = 0, 0, 2, 0.5, 1
-    z2 = 2, 1, 1, 2.5, 1.1
+    z1 = 2  # Separate test for z2<z1, returns negative value with warning
+    z2 = 1
+    results = -969.34452994 *u.Mpc
+    with pytest.warns(AstropyUserWarning):
```

Comment:
Partial warning message match will ensure that this warning is not coming from somewhere else. I didn't include full warning because I want to avoid regex but you are free to make it stricter if you wish:
```suggestion
    with pytest.warns(AstropyUserWarning, match='less than first redshift'):
```